### PR TITLE
feat/add resources for aks

### PR DESCRIPTION
**Pull request by YiscahLevySilas1**
Add `ListEntitiesForPolicies` and `PolicyVersion` resources for AKS

Will be used in controls of CIS-AKS - [5.1.2](https://cyberarmor-io.atlassian.net/browse/SUB-553) and [5.1.3](https://cyberarmor-io.atlassian.net/browse/SUB-554)

### CODE REVIEW

File: cloudsupport/v1/cloudproviderv1.go [hunk 1]
```diff
@@ -253,3 +285,40 @@ func GetPolicyVersionEKS(eksSupport IEKSSupport, cluster string, region string)
 
 	return listPolicyInfo, nil
 }
+
```
Comment (alegrey91) [inline cloudsupport/v1/cloudproviderv1.go:288]: please add **godoc** to describe the `GetPolicyVersionAKS` function

**Review by alegrey91 — CHANGES_REQUESTED** [review]
Code implementation looks good. Just add the **godoc**s on top of new functions.

**Comment by dwertent** [comment]
@YiscahLevySilas1 Please open this PR against the `main` branch 
We are deprecating the `master` branch in a few days.

File: cloudsupport/cloudproviderconfiguration.go [hunk 2]
```diff
@@ -145,8 +145,19 @@ func GetListEntitiesForPoliciesFromCloudProvider(cluster string, cloudProvider s
 		//TODO - implement GKE support
 		return nil, fmt.Errorf(cloudsupportv1.NotSupportedMsg)
 	case cloudsupportv1.AKS:
-		//TODO - implement AKS support
-		return nil, fmt.Errorf(cloudsupportv1.NotSupportedMsg)
+		aksSupport := cloudsupportv1.NewAKSSupport()
```
Comment (dwertent) [inline cloudsupport/cloudproviderconfiguration.go:148]: Can we add unitests?

File: cloudsupport/v1/cloudproviderv1.go [hunk 3]
```diff
@@ -95,6 +95,38 @@ func GetListEntitiesForPoliciesEKS(eksSupport IEKSSupport, cluster string, regio
 	return listEntitiesForPoliciesInfo, nil
 }
 
+func GetListEntitiesForPoliciesAKS(aksSupport IAKSSupport, cluster string, subscriptionId string, resourceGroup string) (*CloudProviderListEntitiesForPolicies, error) {
```
Comment (dwertent) [inline cloudsupport/v1/cloudproviderv1.go:98]: Please add unitests

**Review by dwertent — CHANGES_REQUESTED** [review]

File: cloudsupport/v1/cloudproviderv1.go (hunk 3 above)
Comment (YiscahLevySilas1) [reply cloudsupport/v1/cloudproviderv1.go:98]: I did :)

File: cloudsupport/cloudproviderconfiguration.go (hunk 2 above)
Comment (YiscahLevySilas1) [reply cloudsupport/cloudproviderconfiguration.go:148]: I added unitests for those we have mock interfaces, this func create a "real" AKSSupport obj which tries getting resources from a cluster so I think we can't here

**Review by alegrey91 — APPROVED** [review]
LGTM

**Review by dwertent — APPROVED** [review]

**Pull request merged**